addons/web_editor/models/field_html_history_revision.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class HtmlHistoryRevision(models.Model):
+    _name = "field.html.history.revision"
+    _description = "Field html History Revision"
+
+    res_field = fields.Char('Related Document Model Field Name',
```

Comment:
Would use a m2o towards ir.model.fields (ir_model_field_id) + a related for the name (ir_model_field_name). That way it is correctly unlinked when removing the field, renaming field does not break revision history, ...

Reply:
I made the change but the drawback is that now we have a query every write on the document that extends `field.html.history.mixin` ( from the call to `_get_versioned_field()` ).
This might is not ideal for performance.

Reply:
I'll revert it. It's a nightmare or access right to `ir.model.fields` when your user doesn't have admin right. So I don't think it is worth the trouble.

Reply:
If you handle it cleanly it works, but indeed you have to use _get and _get_ids that are tailored for this kind of usage. Anyway those fields should never be accessed by any other people than admin (and using specific getter in code when filling the table is the right way to do it).

Reply:
ok, but then it cannot be fully "dynamic" with the revision being linked to an `ir.model.field` with a `Many2One` right ? 

And thus we loose one of the benefits of this approach.

Reply:
Don't understand... you can use a m2o, don't see what are the issues. People won't create revisions by hand, meaning they don't have to gain access to 'ir model field' model. Moreover revision is a technical model so using technical fields makes sense, instead of loosely coupling them using char fields. You can also put a group on the field, cleanly check who is writing on field_id, ... it is way cleaner in my opinion, but a bit more complex technically indeed.

Reply:
I had a lot of issue when doing some ORM call from the OWL component (`historyDialog`) because the access rules throw me an error every time I wanted to read the revision data if the revision had a link to `ir.model.view`, So I would have to create proxies method in the model for all those.

Same for the `history_revision_ids` fieldDependencies in knowledge's `topbar.js`. faced some issue with acces right.

But maybe I didn't do it properly.

Reply:
You can use "field_name" who is a related on field_id.name when doing searches. This way we have a correct link at model level (using m2o) and an interface for users (char field). This is generally what we do also for models for example: a link to ir.model but consistency, and a related for the model name usage (domains, user searches, ...). That way you don't have to manually handle renaming or removal of ir.model(.fields), it is automatically updated using the m2o link and the related is there for easy ysage.